crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1514,6 +1563,7 @@ mod tests {
     enum Ty {
         Never,
         Unknown,
+        None,
```

Comment:
There might have been a good reason why `None` was missing so far?

Reply:
We plan to remove `Type::None` but I don't think this is relevant to the test code changes here

https://github.com/astral-sh/ruff/issues/13670

Reply:
Yeah, this is fine